File: src/큐_덱/b_1966.java
package 큐_덱;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class b_1966 {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        Queue<Integer> queue = new LinkedList<>();
        for(int i=0;i<T;i++){
            queue.clear();
            st = new StringTokenizer(br.readLine());
            int n = Integer.parseInt(st.nextToken());
            int index = Integer.parseInt(st.nextToken());
            int list[] = new int[10];
            st = new StringTokenizer(br.readLine());
            for(int j=0;j<n;j++){
                int temp = Integer.parseInt(st.nextToken());
                list[temp]++;
                queue.add(temp);
            }
            int count =0;
            while(true){
                boolean isPrint = true;
                int num = queue.remove();
                for(int j=num+1;j<10;j++){
                    if(list[j]>0){
                        queue.add(num);
                        if(index==0) index = queue.size()-1;
                        else index--;
                        isPrint = false;
                        break;
                    }
                }
                if(isPrint){
                    list[num]--;
                    count++;
                    if(index==0){
                        sb.append(count).append("\n");
                        break;
                    }
                    index--;
                }
            }
        }
        System.out.println(sb);
    }
}

File: src/큐_덱/b_1966.kt
package 큐_덱

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main(args: Array<String>){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val T = br.readLine().toInt()
    val queue: Queue<Int> = LinkedList()
    var st: StringTokenizer
    val sb = StringBuilder()
    for(i in 0 until T){
        st = StringTokenizer(br.readLine())
        val n = st.nextToken().toInt()
        var index = st.nextToken().toInt()
        val array = Array(10){0}
        queue.clear()
        var count = 0
        st = StringTokenizer(br.readLine())
        for(i in 0 until n){
            val num = st.nextToken().toInt()
            array[num]++
            queue.add(num)
        }

        loop@while(true){
            val outNum = queue.remove()
            for(i in outNum+1 until array.size){
                if(array[i]>0){
                    queue.add(outNum)
                    if(index==0) index = queue.size-1
                    else index--
                    continue@loop
                }
            }
            array[outNum]--
            count++
            if(index==0){
                sb.append(count).append("\n")
                break
            }
            index--
        }
    }
    println(sb)
}